File: 2817-minimum-cost-to-make-all-characters-equal/2817-minimum-cost-to-make-all-characters-equal.cpp
class Solution {
public:
    long long minimumCost(string s) {
        int n = s.size();
        long long totalCost = 0;
        
        for (int i = 1; i < n; i++) {
            if (s[i] != s[i - 1]) {
                totalCost += min(i, n - i); // Choose min flip cost
            }
        }
        
        return totalCost;
    }
};

File: 2817-minimum-cost-to-make-all-characters-equal/2817-minimum-cost-to-make-all-characters-equal.java
class Solution {
    public long minimumCost(String s) {
        int n = s.length();
        long totalCost = 0;
        
        for (int i = 1; i < n; i++) {
            if (s.charAt(i) != s.charAt(i - 1)) {
                totalCost += Math.min(i, n - i); // Choose min flip cost
            }
        }
        
        return totalCost;
    }
}